include/envoy/network/filter.h

```diff
@@ -403,6 +403,11 @@ class UdpReadFilterCallbacks {
    * @return the udp listener that owns this read filter.
    */
   virtual UdpListener& udpListener() PURE;
+
+  /**
+   * @return the udp listener is valid or not.
+   */
+  virtual bool isValidUdpListener() PURE;
```

Comment:
Did you consider not to add interface but to change `udpListener` to return optional reference?

Reply:
> Did you consider not to add interface but to change `udpListener` to return optional reference?

Yes. I also considered about that but there are some pros and cons.
Followings are options that I considered.

1. Change the udpListener's signature to return a pointer not reference.
  pros : It is easy to check the return value of the method is null or not so added code is less than current patch set.
  cons : Need to change every code that call the udpListener and need to change the api's signature that can cause compatibility issues.
            The udpListener ensure that the returned value does not have null(I think that is an intention but actually dose not. :( ) so I need to add some code for nullity checks where the udpListener is called.

2. Change the udpListener's signature to return a optional reference.
  Same as first one.

3. Add a new interface to check the nullity.
  pros : It does not touch any previous behaviors so there are no compatibility  issues and integration issues.
            As I mentioned that the udpListener returns reference so that it ensures returned value is not a null but in the some cases does not. I think that the returned value of udpListener almost has valid value. So, it is better to check if returned value is null or not in special cases, I think.
  cons : As I mentioned, a new method should be added in special case.


So, I choose the third option.
Please give your opinions if there are any good options and fix my assumption if it is wrong. :)

Reply:
Got it clearly. thanks! 🙏

Reply:
An optional reference isn't allowed (by the language) unfortunately. I think option 1 is the correct choice. There's not that much code to change, and I'm not worried about compatbility.

Reply:
> An optional reference isn't allowed (by the language) unfortunately. I think option 1 is the correct choice. There's not that much code to change, and I'm not worried about compatbility.

Ok. I got your point. I will revise my patch. :)